frontend/src/views/Quiz.vue

```diff
@@ -14,85 +11,49 @@ import Loader from '@/components/Loader.vue';
 
 const route = useRoute();
 
-const quiz: Ref<Quiz | null> = ref(null);
-const errorOccurred: Ref<boolean> = ref(false);
-const isFetchingQuiz: Ref<boolean> = ref(true);
-
-useFetch<Quiz>(`/api/quizzes/${route.params.id}`)
-    .then((response) => {
-        if (response.error.value) {
-            errorOccurred.value = true;
-            log(response.error.value);
-        } else {
-            quiz.value = response.data.value;
-        }
-        isFetchingQuiz.value = false;
-    })
-    .catch((error) => {
-        errorOccurred.value = true;
-        isFetchingQuiz.value = false;
-        log(error);
-    });
-
-const questionItems: ComputedRef<
-    Record<Quizoot.QuestionItem['question_id'], Quizoot.QuestionItem>
-> = computed(() => {
-    const questionItems: Record<
-        Quizoot.QuestionItem['question_id'],
-        Quizoot.QuestionItem
-    > = {};
-    for (const questionItem of quiz.value?.questions || []) {
-        questionItems[questionItem.question_id] = questionItem;
-    }
-    return questionItems;
+const {
+    data: quiz,
+    isFetching,
+    error,
+} = useFetch<Quizoot.Quiz>(`/api/quizzes/${route.params.id}`);
+
+const questions = computed(() => {
+    return quiz.value?.questions ?? [];
 });
 
-const currentQuestionItem: Ref<Quizoot.QuestionItem | null> = ref(null);
-const currentQuestionRank: Ref<number> = ref(0);
+const currentQuestion = ref(-1);
 
-function getFirstQuestionItem() {
-    for (const question of quiz.value?.questions || []) {
-        if (question.prev_question_id === null) {
-            return question;
-        }
-    }
-    return null;
+function getCurrentQuestion() {
+    return questions.value[currentQuestion.value];
 }
 
 function startQuiz() {
-    currentQuestionItem.value = getFirstQuestionItem();
-    currentQuestionRank.value = 1;
+    currentQuestion.value = 0;
 }
 
 function quitQuiz() {
```

Comment:
this does nothing I thing. We might want to use the router and go back to `/quizzes`

Reply:
Okay.